### ISSUE TITLE
unused-import false positive when using type union in annotations

### ISSUE THREAD

**Issue by Why-not-now** [pylint-dev/pylint#8831]
### Bug description

When using string annotation, the string should encompass the whole annotation, which is needed for . I have not found a workaround over this issue.

Example code:
```py
# a.py

# pylint: disable=missing-docstring
from typing import Literal


def false_unused_imports() -> "int | Literal[True]":
    return True


# def annotation_error() -> int | "Literal[True]":
#     return True
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a.py:4:0: W0611: Unused Literal imported from typing (unused-import)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)
```

### Expected behavior

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

`(no error is found)`

### Pylint version

```shell
pylint 2.17.4
astroid 2.15.5
Python 3.11.4 (tags/v3.11.4:d2340ef, Jun  7 2023, 05:45:37) [MSC v.1934 64 bit (AMD64)]
```

**Comment by Pierre-Sassoulas** [comment]
I suppose the actual use case wouldn't work without the string annotation ? 

```python
def no_string_annotation() -> int | Literal[True]:
    return True
```

Something like this ?

```python
# pylint: disable=missing-docstring
from typing import Literal

class StringAnnotationRequired:
    def false_unused_imports() -> "int | Literal[StringAnnotationRequired]":
        return self

    def annotation_error() -> int | "Literal[StringAnnotationRequired]":
        return self
```

**Comment by Why-not-now** [comment]
Yes that is correct.

**Comment by nickdrozd** [comment]
Adding `from __future__ import annotations` removes the need for any string annotations:

```python
from __future__ import annotations

from typing import Literal
from enum import Enum

class Result(Enum):
    PASS = 0
    FAIL = 1

class MyClass:
    value = 0

    def no_problem(self) -> MyClass | Literal[Result.FAIL]:
        if self.value < 0:
            return self
        return Result.FAIL
```

I wish they would just make this standard behavior so that the future import wouldn't be requrired.

In the meantime, maybe Pylint could suggest adding it wherever it would fix problems?

**Comment by Pierre-Sassoulas** [comment]
There's [still discussion about it](https://discuss.python.org/t/type-annotations-pep-649-and-pep-563/11363/2), the ``__future__`` might not be the future and is still required in python 3.11 to activate the behavior. I would be in favor of keeping the status quo until PEP649/PEP563 are integrated without the need for ``__future__`` or at least a clear consensus is reached in CPython.

**Comment by nickdrozd** [comment]
Looks like [PEP 649 is scheduled for 3.13](https://discuss.python.org/t/pep-649-deferred-evaluation-of-annotations-tentatively-accepted/21331/43). So plain annotations will be eventually become standard. A new check could be added to flag string annotations.
